src/containerapp-preview/azext_containerapp_preview/_utils.py

```diff
@@ -0,0 +1,93 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from azure.cli.core.azclierror import NoTTYError
+from azure.cli.core.extension.dynamic_install import _get_extension_use_dynamic_install_config
+from knack.prompting import prompt_y_n, NoTTYException
+from knack.util import CLIError
+from knack.log import get_logger
+
+from ._constants import MIN_GA_VERSION
+
+logger = get_logger(__name__)
+
+
+def _get_or_add_extension(cmd, extension_name):
+    from azure.cli.core.extension import (
+        ExtensionNotInstalledException, get_extension)
+    from packaging.version import parse
+
+    # this confirmation step may block the automation scripts, because not all customers allow installing extensions without prompt
+    prompt_ext = True
+    try:
+        ext = get_extension(extension_name)
+        # Check extension version
+        # If the extension is automatically upgraded in the context of a command, it needs to reload all the files in the new extension, otherwise it will not find some dependent resources.
+        if ext and parse(ext.version) < parse(MIN_GA_VERSION):
+            msg = f"The command requires the latest version of extension {extension_name}. Run 'az extension add --upgrade -n {extension_name}' to upgrade extension"
```

Comment:
latest version? according to the condition, >= MIN_GA_VERSION is fine

Reply:
update:
`msg = f"The command requires the version of {extension_name} >= {MIN_GA_VERSION}. Run 'az extension add --upgrade -n {extension_name}' to upgrade extension"`